frontend/src/scenes/max/maxThreadLogic.tsx

```diff
@@ -0,0 +1,563 @@
+        stopGeneration: true,
+        completeThreadGeneration: true,
+        addMessage: (message: ThreadMessage) => ({ message }),
+        replaceMessage: (index: number, message: ThreadMessage) => ({ index, message }),
+        setThread: (thread: ThreadMessage[]) => ({ thread }),
+        setMessageStatus: (index: number, status: MessageStatus) => ({ index, status }),
+        retryLastMessage: true,
+        setConversation: (conversation: Conversation) => ({ conversation }),
+        resetThread: true,
+        setTraceId: (traceId: string) => ({ traceId }),
+    }),
+
+    reducers(({ props }) => ({
+        conversation: [
+            props.conversation ? removeConversationMessages(props.conversation) ?? null : null,
+            {
+                setConversation: (_, { conversation }) => conversation,
+            },
+        ],
+
+        threadRaw: [
+            (props.conversation?.messages ?? []) as ThreadMessage[],
+            {
+                addMessage: (state, { message }) => [...state, message],
+                replaceMessage: (state, { message, index }) => [
+                    ...state.slice(0, index),
+                    message,
+                    ...state.slice(index + 1),
+                ],
+                setMessageStatus: (state, { index, status }) => [
+                    ...state.slice(0, index),
+                    {
+                        ...state[index],
+                        status,
+                    },
+                    ...state.slice(index + 1),
+                ],
+                resetThread: (state) => state.filter((message) => !isReasoningMessage(message)),
+                setThread: (_, { thread }) => thread,
+            },
+        ],
+
+        // Specific case when the conversation is in progress on the backend, but the device doesn't have an open stream
+        conversationLoading: [
+            false,
+            {
+                setConversation: (_, { conversation }) =>
+                    conversation && conversation.status === ConversationStatus.InProgress,
+            },
+        ],
+
+        streamingActive: [
+            false,
+            {
+                askMax: () => true,
+                completeThreadGeneration: () => false,
+            },
+        ],
+
+        // Trace ID is used for the conversation metrics in the UI
+        traceId: [null as string | null, { setTraceId: (_, { traceId }) => traceId, cleanThread: () => null }],
+    })),
+
+    listeners(({ actions, values, cache, props }) => ({
+        askMax: async ({ prompt, generationAttempt }, breakpoint) => {
+            // Clear the question
+            actions.setQuestion('')
+            // Set active streaming threads, so we now how many are running
+            actions.setActiveStreamingThreads(1)
+
+            // For a new conversations, set the temporary conversation ID, which will be replaced with the actual conversation ID once the first message is generated
```

Comment:
We can also perhaps generate the conversation UUID on the client side. I don't think there's any downside to that – the benefit is that we'd init `maxThreadLogic()` with the _actual_ ID right out of the gate, no dummy temporary one in between.